### DISCUSSION TITLE
Support multiple messages in protobuf formatted Kafka sources

### DISCUSSION THREAD

**Discussion by chaas** [MaterializeInc/materialize#29603] [Feature Requests]
Currently Kafka sources that carry protobuf messages require that the protobuf schema file specified describe exactly one protobuf message.
Make this more flexible such that they can support a flexible number (0 or mulitple).

**Comment by juanAngel** [comment]
I had the same problem, and it seems the restriction only applies when you use a schema record.

When you use an embedded schema in the query itself, this problem doesn't exist, since one of the parameters specifies the message to use.
